sycl/test-e2e/GroupAlgorithm/root_group.cpp

```diff
@@ -53,10 +56,15 @@ void testRootGroup() {
     sycl::accessor data{dataBuf, h};
     h.parallel_for<
         class RootGroupKernel>(range, props, [=](sycl::nd_item<1> it) {
+      volatile float X = 1.0f;
+      volatile float Y = 1.0f;
       auto root = it.ext_oneapi_get_root_group();
       data[root.get_local_id()] = root.get_local_id();
       sycl::group_barrier(root);
-
+      if (it.get_group(0) % 2 == 0) {
+        X += sycl::sin(X);
```

Comment:
I don't see that we ever check neither `X` nor `Y`. Should we? Otherwise, why we need this? How do we make sure it ran?

Reply:
I wanted to explicitly delay some of the workgroups by adding them more work to do, because I've seen this test passing if insufficient barrier was used. For instance on CUDA backend, doing work-group wide barrier would be enough for it to pass and that is not correct. I think this test should perform some work-group divergence to actually check that we actually perform *gpu-wide* barrier.

> How do we make sure it ran?

The `X` and `Y` are declared as `volatile` and my understanding was that this would prevent compiler from removing them with some optimization.